devito/iteration.py

```diff
@@ -1,38 +1,50 @@
 from collections import Iterable
+from itertools import chain
 
 import cgen
-from devito.codeprinter import ccode
+
+from devito.expression import Expression
+from devito.tools import filter_ordered
 
 __all__ = ['Iteration']
 
 
-class Iteration(object):
+class Iteration(Expression):
```

Comment:
So that I can type check every item in `Propagator/StencilKernel.stencils`. They both should provide the same interface functions, like `ccode` and `susbtitute`.